lib/llm/src/http/client.rs

```diff
@@ -64,6 +64,8 @@ pub struct HttpRequestContext {
     created_at: Instant,
     /// Whether the request has been stopped
     stopped: Arc<std::sync::atomic::AtomicBool>,
+    /// Child contexts to be stopped if this is stopped
+    child_context: Arc<Mutex<Vec<Arc<dyn AsyncEngineContext>>>>,
```

Comment:
nit: That inner `Arc` (shared ownership) could probably be a `Box` (single owner), because it's behind a Mutex.